vector/src/main/java/im/vector/app/features/matrixto/MatrixToBottomSheetViewModel.kt

```diff
@@ -0,0 +1,192 @@
+import im.vector.app.R
+import im.vector.app.core.extensions.exhaustive
+import im.vector.app.core.platform.VectorViewModel
+import im.vector.app.core.resources.StringProvider
+import im.vector.app.features.raw.wellknown.getElementWellknown
+import im.vector.app.features.raw.wellknown.isE2EByDefault
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.matrix.android.sdk.api.extensions.tryOrNull
+import org.matrix.android.sdk.api.raw.RawService
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.permalinks.PermalinkData
+import org.matrix.android.sdk.api.session.permalinks.PermalinkParser
+import org.matrix.android.sdk.api.session.profile.ProfileService
+import org.matrix.android.sdk.api.session.room.model.create.CreateRoomParams
+import org.matrix.android.sdk.api.session.user.model.User
+import org.matrix.android.sdk.api.util.JsonDict
+import org.matrix.android.sdk.api.util.toMatrixItem
+import org.matrix.android.sdk.internal.util.awaitCallback
+
+class MatrixToBottomSheetViewModel @AssistedInject constructor(
+        @Assisted initialState: MatrixToBottomSheetState,
+        @Assisted val args: MatrixToBottomSheet.MatrixToArgs,
+        private val session: Session,
+        private val stringProvider: StringProvider,
+        private val rawService: RawService) : VectorViewModel<MatrixToBottomSheetState, MatrixToAction, MatrixToViewEvents>(initialState) {
+
+    @AssistedInject.Factory
+    interface Factory {
+        fun create(initialState: MatrixToBottomSheetState,
+                   args: MatrixToBottomSheet.MatrixToArgs): MatrixToBottomSheetViewModel
+    }
+
+    init {
+        setState {
+            copy(matrixItem = Loading())
+        }
+        viewModelScope.launch(Dispatchers.IO) {
+            resolveLink()
+        }
+    }
+
+    private suspend fun resolveLink() {
+        when {
+            args.matrixToLink != null -> {
+                val linkedId = PermalinkParser.parse(args.matrixToLink)
+                if (linkedId is PermalinkData.FallbackLink) {
+                    setState {
+                        copy(
+                                matrixItem = Fail(IllegalArgumentException(stringProvider.getString(R.string.permalink_malformed))),
+                                startChattingState = Uninitialized
+                        )
+                    }
+                    return
+                }
+
+                when (linkedId) {
+                    is PermalinkData.UserLink -> {
+                        val user = resolveUser(linkedId.userId)
+                        setState {
+                            copy(
+                                    matrixItem = Success(user.toMatrixItem()),
+                                    startChattingState = Success(Unit)
+                            )
+                        }
+                    }
+                    is PermalinkData.RoomLink -> TODO()
```

Comment:
The `TODO()` will make the app crash...

Reply:
yes but not called yet, anyhow will log silently instead